components/core/src/Grep.cpp

```diff
@@ -670,35 +696,59 @@ Grep::get_bounds_of_next_potential_var (const string& value, size_t& begin_pos,
+                log_surgeon::ParserInputBuffer parser_input_buffer;
+                if (has_suffix_wildcard) { //text*
+                    // TODO: creating a string reader, setting it equal to a 
+                    //  string, to read it into the ParserInputBuffer, seems
+                    //  like a convoluted way to set a string equal to a string,
+                    //  should be improved when adding a SearchParser to 
+                    //  log_surgeon
+                    string_reader.open(value.substr(begin_pos, end_pos - begin_pos - 1));
+                    parser_input_buffer.read_if_safe(reader_wrapper);
+                    forward_lexer.reset();
+                    forward_lexer.scan_with_wildcard(
+                            parser_input_buffer,
+                            value[end_pos - 1],
+                            search_token
+                    );
+                } else if (has_prefix_wildcard) { // *text
+                    std::string value_reverse
+                            = value.substr(begin_pos + 1, end_pos - begin_pos - 1);
+                    std::reverse(value_reverse.begin(), value_reverse.end());
+                    string_reader.open(value_reverse);
+                    parser_input_buffer.read_if_safe(reader_wrapper);
+                    reverse_lexer.reset();
+                    reverse_lexer.scan_with_wildcard(
+                            parser_input_buffer,
+                            value[begin_pos],
+                            search_token
+                    );
+                } else { // no wildcards
+                    string_reader.open(value.substr(begin_pos, end_pos - begin_pos));
+                    parser_input_buffer.read_if_safe(reader_wrapper);
+                    forward_lexer.reset();
+                    forward_lexer.scan(parser_input_buffer, search_token);
+                    search_token.m_type_ids_set.insert(search_token.m_type_ids_ptr->at(0));
                 }
-            } else { // no wildcards
-                StringReader stringReader;
-                stringReader.open(value.substr(begin_pos, end_pos - begin_pos));
-                forward_lexer.reset(stringReader);
-                compressor_frontend::Token token = forward_lexer.scan();
-                if (token.m_type_ids->at(0) != (int) compressor_frontend::SymbolID::TokenUncaughtStringID &&
-                    token.m_type_ids->at(0) != (int) compressor_frontend::SymbolID::TokenEndID) {
+                // TODO: use a set so its faster
+                // auto const& set = search_token.m_type_ids_set;
+                // if (set.find(static_cast<int>(log_surgeon::SymbolID::TokenUncaughtStringID))
+                //            == set.end()
+                //     && set.find(static_cast<int>(log_surgeon::SymbolID::TokenEndID))
+                //            == set.end())
+                // {
+                //     is_var = true;
+                // }
```

Comment:
Is this supposed to be commented out?

Reply:
Yea cause the set isn't implemented in the current implementation of SearchToken. But, since we're changing search stuff separately, to address bigger functional and performance issues, I'm leaving this to be handled with that change.